api4/role.go

```diff
@@ -125,14 +130,45 @@ func patchRole(c *Context, w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if patch.Permissions != nil {
+		deltaPermissions := model.PermissionsChangedByPatch(oldRole, patch)
+
+		notAllowedPermissions := []string{
+			model.PERMISSION_SYSCONSOLE_WRITE_USERMANAGEMENT_SYSTEM_ROLES.Id,
+			model.PERMISSION_SYSCONSOLE_READ_USERMANAGEMENT_SYSTEM_ROLES.Id,
+			model.PERMISSION_MANAGE_ROLES.Id,
+		}
```

Comment:
Create as const, it doesn't need created each time through.

Reply:
Do you mean drag it out to a global scope and use a `var` ?  I took them out to a global scope and instead of using `:=` used `=` making them somewhat of a pseudo-constant.